src/olympia/versions/views.py

```diff
@@ -18,25 +18,15 @@
 from olympia.versions.models import Version
 
 
-# The version detail page redirects to the version within pagination, so we
-# need to enforce the number of versions per page.
-PER_PAGE = 30
-addon_view = addon_view_factory(Addon.objects.valid)
-
 log = olympia.core.logger.getLogger('z.versions')
 
 
-@addon_view
+@addon_view_factory(lambda: Addon.objects.public().no_transforms())
 @non_atomic_requests
 def update_info(request, addon, version_num):
-    version = Version.objects.filter(
-        addon=addon,
-        version=version_num,
-        file__status__in=amo.VALID_FILE_STATUSES,
-        channel=amo.RELEASE_CHANNEL_LISTED,
-    ).last()
-    if not version:
-        raise http.Http404()
+    version = get_object_or_404(
+        addon.versions.reviewed().only_translations(), version=version_num
```

Comment:
We've lost the `channel` filter.

Reply:
Good catch. I'll add another test to make sure we can't see unlisted data